docs/dev/protocols.md

```diff
@@ -149,46 +149,35 @@ The CQL protocol support can be disabled altogether by setting the
 
 These option names were chosen for backward-compatibility with Cassandra
 configuration files: they refer to CQL as the "native transport", to
-contrast with the older Thrift protocol (described below) which wasn't
-native to Cassandra.
+contrast with the older Thrift protocol which wasn't
+native to Cassandra. The thrift protocol was once supported by Scylla,
+but the support of this protocol was later deprecated and removed.
 
 There is also a `rpc_address` configuration option to set the IP address
 (and therefore network interface) on which Scylla should listen for the
 CQL protocol. This address defaults to `localhost`, but in any setup except
-a one-node test, should be overridden. Note that the same option `rpc_address`
-applies to both CQL and Thrift protocols.
-
-TODO: there is also `rpc_interface` option... Which wins? What's the default?
-
-## Thrift client protocol
+a one-node test, should be overridden.
 
 The Apache Thrift protocol was early Cassandra's client protocol, until
 it was superseded in Cassandra 1.2 with the binary CQL protocol. Thrift
 was still nominally supported by both Cassandra and Scylla for many years,
-but was recently dropped in Cassandra (version 4.0) and is likely to be
-dropped by Scylla in the future as well, so it is not recommended for new
-applications.
+but was recently dropped in Cassandra (version 4.0) and was also
+dropped by Scylla in version 5.5 as well.
 
-By default, Scylla does not enable the Thrift server. In order to use it,
-it must be explicitly enabled by setting the `start_rpc` configuration option
-to true.
-
-When Thrift is enabled, by default scylla listens to the Thrift protocol on port 9160,
+When Thrift was enabled by Scylla versions earlier than 5.5, by default scylla
+listens to the Thrift protocol on port 9160,
 which can be configured via the `rpc_port` configuration option. Again, this confusing
 name was used for backward-compatibility with Cassandra's configuration files.
 Cassandra used the term "rpc" because Apache Thrift is a remote procedure
 call (RPC) framework. In Scylla, this name is especially confusing, because
 as mentioned above, Scylla's internal communication protocol is based on
 Seastar's RPC, which has nothing to do with the "`rpc_port`" described here.
 
-There is also a `rpc_address` configuration option to set the IP address
-(and therefore network interface) on which Scylla should listen for the
-Thrift protocol. This address defaults to `localhost`, but in any
-setup except a one-node test, should be overridden. Note that the same
-option `rpc_address` applies to both CQL and Thrift protocols.
+This option still stays with us, because scylla need to be able to consume
```

Comment:
really? I thought you actually removed rpc_port - and it's the rpc_address option (whose description you removed from this document....) which remains?

I think there's some mixup here, or I'm misunderstanding something.

Reply:
it was a mistake to remove them in `db/config.cc` and `db/config.hh`. i intended to keep them, but to mark them `Deprecated` before finally removing them. as we should do the `Used` -> `Deprecated` -> `<Unused>` -> `<removed>` dance. as these option could still be sitting in users' `scylla.yaml` file. we need to warn users, so that they can remove them, before removing these options in another major release.

in the latest revision, i am putting `rpc_port` and `start_rpc` back.